includes/api/class-wc-admin-rest-onboarding-profile-controller.php

```diff
@@ -0,0 +1,300 @@
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Update onboarding profile data.
+	 *
+	 * @param  WP_REST_Request $request Request data.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function update_items( $request ) {
+		$query_args      = $this->prepare_objects_query( $request );
+		$onboarding_data = get_option( 'wc_onboarding_profile', array() );
+		$update          = update_option( 'wc_onboarding_profile', array_merge( $onboarding_data, $query_args ) );
+
+		if ( $update ) {
+			$result = array(
+				'status'  => 'success',
+				'message' => __( 'Onboarding profile data has been updated.', 'woocommerce-admin' ),
+			);
+		} else {
+			$result = array(
+				'status'  => 'error',
+				'message' => __( 'There was an error updating the onboarding profile data.', 'woocommerce-admin' ),
+			);
+		}
+
+		$response = $this->prepare_item_for_response( $result, $request );
+		$data     = $this->prepare_response_for_collection( $response );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepare objects query.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return array
+	 */
+	protected function prepare_objects_query( $request ) {
+		$args       = array();
+		$properties = self::get_profile_properties();
+
+		foreach ( $properties as $key => $property ) {
+			if ( isset( $request[ $key ] ) ) {
+				$args[ $key ] = $request[ $key ];
+			}
+		}
+
+		/**
+		 * Filter the query arguments for a request.
+		 *
+		 * Enables adding extra arguments or setting defaults for a post
+		 * collection request.
+		 *
+		 * @param array           $args    Key value array of query var to query value.
+		 * @param WP_REST_Request $request The request used.
+		 */
+		$args = apply_filters( 'woocommerce_rest_onboarding_profile_object_query', $args, $request );
+
+		return $args;
+	}
+
+
+	/**
+	 * Prepare the data object for response.
+	 *
+	 * @param object          $item Data object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response $response Response data.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$data     = $this->add_additional_fields_to_object( $item, $request );
+		$data     = $this->filter_response_by_context( $data, 'view' );
+		$response = rest_ensure_response( $data );
+
+		/**
+		 * Filter the list returned from the API.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param array            $item     The original item.
+		 * @param WP_REST_Request  $request  Request used to generate the response.
+		 */
+		return apply_filters( 'woocommerce_rest_prepare_onboarding_profile', $response, $item, $request );
+	}
+
+	/**
+	 * Get onboarding profile properties.
+	 *
+	 * @return array
+	 */
+	public static function get_profile_properties() {
+		$properties = array(
+			'account_type'   => array(
+				'type'        => 'string',
+				'description' => __( 'Account type used for Jetpack.', 'woocommerce-admin' ),
```

Comment:
What's the account type setting for? Would this ever be anything other than just `jetpack` seen in your screenshot?

Reply:
The account type can be `new`, `existing`, or `google` if they choose to connect their Google account.  If you think this is property is unnecessary, let me know and I'll remove.

Reply:
Gotcha! I think that could be valuable to track/know so we should keep it 👍.